### DISCUSSION TITLE
Where some idea of feature I need

### DISCUSSION THREAD

**Discussion by shiipou** [zekroTJA/shinpuru#324] [Ideas]
I post you some idea of feature I need for my server, If I find time I do PR for some, but not before many week.

- Auto voice channel 
  > When a user join a voice-channel named for example "create-new-channel", the bot will create two voice channel and move the user to one of these new voice channel, the second will be a waiting room, from anyone can ask to join the main voice-channel if owner accept the request. This user will be granted all permission on this channel. if configured, you can link a text-channel to this voice-channel. If nobody left in the voice channel, the bot will delete the both voice and text channel. Text channel must be seen only if user is in the voice channel.
- Role selection using message component or reaction role.
  > When a user need a specific public role, he can select it just by clicking a button on a message posted by an admin.
- Log user playtime for only a list of game, that can be a top list of player (Top in time of play) for the game of our community.

**Comment by zekroTJA** [comment]
First of all, thank you very much for sharing your ideas and your general interest into the project. 💖

I want to give some of my thoughts about your ideas:

> Auto voice channel

I actually had this exact feature implemented in one of my very first bots. So yeah, would be a nice Idea to have it in shinpuru as well. If you want to contribute that feature, I would suggest creating a new slash command (`internal/slashcommands`) with the following sub commands:
- `register <channel>` - Register an existing voice channel as auto channel.
- `remove <channel>` - Remove a registered channel from the list of auto channels.
- `list` - List all registered auto channels on the guild.

The command should have the permission domain `sp.guild.config.autochannel`. It would be also neat when the listener for creating auto channels would also check for a permission like `sp.chat.autochannel` or something like that.

> Role selection using message component or reaction role.

I often had requests to such a feature and I actually think that there are a lot of bots which exclusively provide such a feature and do that very well. I am not a huge fan of taking core features of other bots and building a rip-off like implementation of that into shinpuru. Also especially when keeping the core concept of the bot in mind.

> Log user playtime for only a list of game, that can be a top list of player (Top in time of play) for the game of our community.

Sadly, such a feature will not be implemented into shinpuru. First of all, this would require logging the presence information of users which is very privacy critical data which I am absolutely not comfortable with and would also lead to a lot of very complex trouble. Also, you need the privileged API intent for capturing presence events which is nearly impossible to get though because of the issues mentioned before. That was especially the reason why I created dgrs - to dodge the presence events for state management.

However, if you want to contribute some of your ideas to shinpuru, feel free to do so. If you need help in any way, don't be afraid to ask me directly via [Discord](https://discord.zekro.de) (`zekro#0001`).   

Also, take a look into the `CONTRIBUTING.md` document where you can find a lot of information about the structure of the project. ;)
https://github.com/zekroTJA/shinpuru/blob/master/CONTRIBUTING.md
